OSSM-9000: Create a default revision tag and relabel namespace (#89561)
openshift/openshift-docs: modules/ossm-creating-a-default-revision-tag-and-relabeling-the-namespaces.adoc

// Module included in the following assemblies:
//
// * service-mesh-docs-main/migrating/cluster-wide/ossm-migrating-cluster-wide-assembly.adoc

:_mod-docs-content-type: PROCEDURE
[id="ossm-creating-a-default-revision-tag-and-relabeling-the-namespaces_{context}"]
= Creating the default revision tag and relabeling the namespaces

This procedure explains how to create the default revision tag and relabel the namespaces after you have completed the {SMProduct} 2 to {SMProduct} 3 cluster-wide migration process using istio injection label.

> The `bookinfo` example application is being used for demonstration purposes with a minimal example for the `Istio` resource. For more information on configuration differences between the {SMProduct} 2 `ServiceMeshControlPlane` resource and the {SMProduct} 3 `Istio` resource, see "Configuration fields mapping between Service Mesh 2 and Service Mesh 3".

Reviewer: ```suggestion
The `bookinfo` example application is being used for demonstration purposes.
```

Reviewer: We are not touching Istio resource here.

Author: Done

.Prerequisites

* You have completed the {SMProduct} 2 to {SMProduct} 3 cluster-wide migration process by using {istio} injection label.

.Procedure

. Create a YAML files called `rev-tag.yml` that defines the `IstioRevisionTag` resource:
+
.Example `IstioRevisionTag` resource
[source,yaml,subs="attributes,verbatim"]
----
apiVersion: sailoperator.io/v1
kind: IstioRevisionTag
metadata:
  name: default
spec:
  targetRef:
    kind: IstioRevision
    name: ossm-3-v1-24-3
----

. Apply the YAML file by running the following command:
+
[source,terminal]
----
$ oc apply -f rev-tag.yaml
----

. Verify the status of the `IstioRevisionTag` resource by running the following command:
+
[source,terminal]
----
$ oc get istiorevisiontags
----
+
.Example output
[source,terminal]
----
NNAME      STATUS                    IN USE   REVISION        AGE
default    NotReferencedByAnything   False    ossm-3-v1-24-3  18s
----

. Add the `istio-injection=enabled` label to the `bookinfo` namespace, and remove the `istio.io/rev` label by running the following command:
+
[source,terminal]
----
$ oc label ns bookinfo istio-injection=enabled istio.io/rev-
----
+
[NOTE]
====

> Repeat this step for each namespace you are migrating.

Reviewer: Repeat steps 4 and 5. Labeling + restart

Author: Done. I think I like it better the way I just changed it.

:_mod-docs-content-type: PROCEDURE
[id="ossm-creating-a-default-revision-tag-and-relabeling-the-namespaces_{context}"]
= Creating the default revision tag and relabeling the namespaces

This procedure explains how to create the default revision tag and relabel the namespaces after you have completed the {SMProduct} 2 to {SMProduct} 3 cluster-wide migration process using istio injection label.

The `bookinfo` example application is being used for demonstration purposes with a minimal example for the `Istio` resource. For more information on configuration differences between the {SMProduct} 2 `ServiceMeshControlPlane` resource and the {SMProduct} 3 `Istio` resource, see "Configuration fields mapping between Service Mesh 2 and Service Mesh 3".

You can follow these same steps with your own workloads.

.Prerequisites

* You have completed the {SMProduct} 2 to {SMProduct} 3 cluster-wide migration process using istio injection label.

.Procedure

. Create a YAML files called `rev-tag.yml` that defines the IstioRevisionTag resource.
+
.Example `IstioRevisionTag` resource
[source,yaml,subs="attributes,verbatim"]
----

> apiVersion: sailoperator.io/v1alpha1

Reviewer: ```suggestion
apiVersion: sailoperator.io/v1
```

Author: Done

Author: Done

default    NotReferencedByAnything   False    ossm-3-v1-24-3  18s
----

. Add `istio-injection=enabled` label and remove `istio.io/rev` label by running the following command:
+
[source,terminal]
----
$ oc label ns bookinfo istio-injection=enabled istio.io/rev-
----

ADD step to repeats steps x to y for each namespace. Delete the step below pertaining to gateways

. If you are using gateways, you must migrate them now. For more information, see "Migrating gateways from Service Mesh 2 to Service Mesh 3".

. Restart the workloads by running the following command:
+
[source,terminal]
----
$ oc rollout restart deployments -n bookinfo
----

. Verify that the `IstioRevisionTag` resource is in use by running the following command:
+
[source,terminal]
----
$ oc get istiorevisiontags
----
+
.Example output
[source,terminal]
----
NAME      STATUS    IN USE   REVISION        AGE
default   Healthy   True     ossm-3-v1-24-3  28s
----

.Verification

. Ensure that expected workloads are managed by the new control plane by running the following command:
+
[source,terminal]
----
$ istioctl ps -n bookinfo
----
+
.Example output
[source,terminal]
----
UPDATE EXAMPLE OUTPUT          
----

> . If the 2.6 installation contains additional namespaces, migrate the next namespace now.

Reviewer: This should be removed when the "step to repeats steps x to y for each namespace" is added

// Module included in the following assemblies:
//
// * service-mesh-docs-main/migrating/cluster-wide/ossm-migrating-cluster-wide-assembly.adoc

:_mod-docs-content-type: PROCEDURE
[id="ossm-creating-a-default-revision-tag-and-relabeling-the-namespaces_{context}"]
= Creating the default revision tag and relabeling the namespaces

> This procedure explains how to create the default revision tag and relabel the namespaces after you have completed the {SMProduct} 2 to {SMProduct} 3 cluster-wide migration process using istio injection label.

Author: Done

> This procedure explains how to create the default revision tag and relabel the namespaces after you have completed the {SMProduct} 2 to {SMProduct} 3 cluster-wide migration process by using the {istio} injection label.

Reviewer 2: ```suggestion
You can create the default revision tag and relabel the namespaces after you have completed the {SMProduct} 2 to {SMProduct} 3 cluster-wide migration process by using the {istio} injection label.
```

> Avoid self-referential language, such as "This topic covers…​" or "Use this procedure to…​".
- https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc

Author: Done

> The `bookinfo` example application is being used for demonstration purposes.

Reviewer 2: ```suggestion
The `bookinfo` application is used as an example.
```

Author: Done

.Prerequisites

* You have completed the {SMProduct} 2 to {SMProduct} 3 cluster-wide migration process by using {istio} injection label.

.Procedure

> . Create a YAML files called `rev-tag.yml` that defines the `IstioRevisionTag` resource:

Reviewer 2: ```suggestion
. Create a YAML file called `rev-tag.yaml` that defines the `IstioRevisionTag` resource:
```

Replace/yml/yaml

Also the `oc apply` command below uses `.yaml`

Author: Updated files to file. Thanks for catching this. In this doc set, we typically use the .yml extension. Both the .yml and .yaml extensions are used in industry. The developers said they see .yml more frequently.

Author: Updated. Done.

.Procedure

. Create a YAML files called `rev-tag.yml` that defines the IstioRevisionTag resource.
+
.Example `IstioRevisionTag` resource
[source,yaml,subs="attributes,verbatim"]
----
apiVersion: sailoperator.io/v1alpha1
kind: IstioRevisionTag
metadata:
  name: default
spec:
  targetRef:
    kind: IstioRevision
    name: ossm-3-v1-24-3
----

. Apply the YAML file by running the following command:
+
[source,terminal]
----
$ oc apply -f rev-tag.yaml
----

. Verify that the status of the IstioRevisionTag resource by running the following command:
+
[source,terminal]
----
$ oc get istiorevisiontags
----
+
.Example output
[source,terminal]
----
NNAME      STATUS                    IN USE   REVISION        AGE
default    NotReferencedByAnything   False    ossm-3-v1-24-1  18s
----

. Add `istio-injection=enabled` label and remove `istio.io/rev` label by running the following command:
+
[source,terminal]
----
$ oc label ns bookinfo istio-injection=enabled istio.io/rev-
----
+
.Example output
[source,terminal]
----
Filip should we show any sort of example output? How will customers know this was completed successfully?
----

> . If you are using gateways, you must migrate them now. For more information, see "Migrating gateways from Service Mesh 2 to Service Mesh 3".

Author: Done

. Restart the workloads by running the following command:
+
[source,terminal]
----
$ oc rollout restart deployments -n bookinfo
----

. Verify that the `IstioRevisionTag` resource is in use by running the following command:
+
[source,terminal]
----
$ oc get istiorevisiontags
----
+
.Example output
[source,terminal]
----
NAME      STATUS    IN USE   REVISION        AGE
default   Healthy   True     ossm-3-v1-24-3  28s
----

.Verification

. Ensure that expected workloads are managed by the new control plane by running the following command:
+
[source,terminal]
----
$ istioctl ps -n bookinfo
----
+
.Example output
[source,terminal]
----

> $ istioctl ps -n bookinfo

Author: @FilipB please paste the output when you have time. You can do so here or in slack if slack if better for formatting reasons.

> $ istioctl ps -n bookinfo

Reviewer: ```
NAME                                         CLUSTER        CDS              LDS              EDS             RDS              ECDS        ISTIOD                                     VERSION
details-v1-79dfbd6fff-t5lzm.bookinfo         Kubernetes     SYNCED (57s)     SYNCED (57s)     SYNCED (3s)     SYNCED (57s)     IGNORED     istiod-ossm-3-v1-24-3-6595bf8695-s8ktn     1.24.3
details-v1-7cb48d8bb-6rjq8.bookinfo          Kubernetes     SYNCED (3s)      SYNCED (3s)      SYNCED (3s)     SYNCED (3s)      IGNORED     istiod-ossm-3-v1-24-3-6595bf8695-s8ktn     1.24.3
productpage-v1-7d9cdf655d-cqk48.bookinfo     Kubernetes     SYNCED (10s)     SYNCED (10s)     SYNCED (3s)     SYNCED (10s)     IGNORED     istiod-ossm-3-v1-24-3-6595bf8695-s8ktn     1.24.3
ratings-v1-5b67b59fcb-w4whk.bookinfo         Kubernetes     SYNCED (18s)     SYNCED (18s)     SYNCED (3s)     SYNCED (18s)     IGNORED     istiod-ossm-3-v1-24-3-6595bf8695-s8ktn     1.24.3
reviews-v1-585fc84dbb-fvm2h.bookinfo         Kubernetes     SYNCED (11s)     SYNCED (11s)     SYNCED (3s)     SYNCED (11s)     IGNORED     istiod-ossm-3-v1-24-3-6595bf8695-s8ktn     1.24.3
reviews-v2-65cb66b45c-6ggp9.bookinfo         Kubernetes     SYNCED (57s)     SYNCED (57s)     SYNCED (3s)     SYNCED (57s)     IGNORED     istiod-ossm-3-v1-24-3-6595bf8695-s8ktn     1.24.3
reviews-v2-698b86b848-v92xq.bookinfo         Kubernetes     SYNCED (3s)      SYNCED (3s)      SYNCED (3s)     SYNCED (3s)      IGNORED     istiod-ossm-3-v1-24-3-6595bf8695-s8ktn     1.24.3
reviews-v3-6cbc49c8c8-v4jck.bookinfo         Kubernetes     SYNCED (11s)     SYNCED (11s)     SYNCED (3s)     SYNCED (11s)     IGNORED     istiod-ossm-3-v1-24-3-6595bf8695-s8ktn     1.24.3
```

> $ istioctl ps -n bookinfo

Author: Done

> $ istioctl ps -n bookinfo

Author: Done

details-v1-7f46897b-d497c.bookinfo            Kubernetes     SYNCED          SYNCED          SYNCED          SYNCED          NOT SENT     istiod-install-istio-system-866b57d668-6lpcr     1.20.8
productpage-v1-74bfbd4d65-vsxqm.bookinfo      Kubernetes     SYNCED (4s)     SYNCED (4s)     SYNCED (3s)     SYNCED (4s)     IGNORED      istiod-ossm-3-v1-24-3-797bb4d78f-xpchx           1.24.3
ratings-v1-559b64556-c5ppg.bookinfo           Kubernetes     SYNCED          SYNCED          SYNCED          SYNCED          NOT SENT     istiod-install-istio-system-866b57d668-6lpcr     1.20.8
reviews-v1-847fb7c54d-qxt5d.bookinfo          Kubernetes     SYNCED          SYNCED          SYNCED          SYNCED          NOT SENT     istiod-install-istio-system-866b57d668-6lpcr     1.20.8
reviews-v2-5c7ff5b77b-8jbhd.bookinfo          Kubernetes     SYNCED          SYNCED          SYNCED          SYNCED          NOT SENT     istiod-install-istio-system-866b57d668-6lpcr     1.20.8
reviews-v3-5c5d764c9b-rrx8w.bookinfo          Kubernetes     SYNCED          SYNCED          SYNCED          SYNCED          NOT SENT     istiod-install-istio-system-866b57d668-6lpcr     1.20.8
----
+

> The previous output shows that the `productpage-v1` deployment is the only deployment that restarted and was injected with the 3.0 proxy. Even if there are different versions of the proxies, communication between services still works.

Author: I've deleted the text but left to syntax for the example output.

> The previous output shows that the `productpage-v1` deployment is the only deployment that restarted and was injected with the 3.0 proxy. Even if there are different versions of the proxies, communication between services still works.

Author: Done
